tests/components/esphome/test_update.py

```diff
@@ -0,0 +1,64 @@
+"""Test ESPHome update entities."""
+from unittest.mock import Mock, patch
+
+import pytest
+
+from homeassistant.components.esphome.dashboard import async_set_dashboard_info
+
+
+@pytest.fixture(autouse=True)
+def stub_reconnect():
+    """Stub reconnect."""
+    with patch("homeassistant.components.esphome.ReconnectLogic.start"):
+        yield
+
+
+@pytest.mark.parametrize(
+    "devices_payload,expected_state,expected_attributes",
+    [
+        (
+            [{"name": "test", "current_version": "1.2.3"}],
+            "on",
+            {"latest_version": "1.2.3", "installed_version": "1.0.0"},
+        ),
+        (
+            [{"name": "test", "current_version": "1.0.0"}],
+            "off",
+            {"latest_version": "1.0.0", "installed_version": "1.0.0"},
+        ),
+        (
+            [],
+            "unavailable",
+            {},
+        ),
+    ],
+)
+async def test_update_entity(
+    hass,
+    mock_config_entry,
+    mock_device_info,
+    devices_payload,
+    expected_state,
+    expected_attributes,
+):
+    """Test ESPHome update entity."""
+    async_set_dashboard_info(hass, "mock-addon-slug", "mock-addon-host", 1234)
+
+    mock_config_entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.esphome.update.DomainData.get_entry_data",
+        return_value=Mock(available=True, device_info=mock_device_info),
+    ), patch(
+        "homeassistant.components.esphome.dashboard.ESPHomeDashboardAPI.get_devices",
+        return_value={"configured": devices_payload},
+    ):
+        assert await hass.config_entries.async_forward_entry_setup(
```

Comment:
Nowadays we set up the config entry, not only forwarding it in tests. This is a legacy test pattern.